docs/docs/adrs/adr-018-permissionless-ics.md

```diff
@@ -0,0 +1,270 @@
+for Top N chains. Because of this, this ADR outlines a solution that attempts to preserve as much of the governance proposal code
+as possible. In what follows, what we describe applies to both governance-proposed consumer (i.e., Top N) chains and transaction-based (i.e., Opt In) chains.
+
+A consumer chain can reside in three phases: i) _prelaunched_, ii) _launched_, and iii) _stopped_ phase as seen
+in the diagram below:
+![Phases of a consumer chain](./adr18_phases_of_a_consumer_chain.png)
+
+When a Top N chain is first proposed through a `ConsumerAdditionProposal` or an Opt In chain is added through a transaction,
+the consumer chain resides in the _prelaunched_ phase. At this state, validators can choose to opt in on the consumer chain. Additionally,
+an Opt In chain can choose to change parameters of the to-be-launched chains, such as `spawnTime`, etc. by issuing a specific transaction.
+This is not the case for Top N chains, where a `ConsumerModificationProposal` can only be issued after a consumer
+chain [has started](https://github.com/cosmos/interchain-security/blob/v4.3.0/x/ccv/provider/keeper/proposal.go#L150).
+
+When the [`spawnTime`](https://github.com/cosmos/interchain-security/blob/v4.3.0/proto/interchain_security/ccv/provider/v1/provider.proto#L55)
+passes and [at least one validator has opted in](https://github.com/cosmos/interchain-security/blob/v4.3.0/x/ccv/provider/keeper/proposal.go#L455)
+the chain can launch and moves to the _launched_ phase. While in launched phase, a Top N consumer chain can choose to modify
+its parameters through a `ConsumerModificationProposal` and an Opt In chain can change its parameters by issuing a transaction.
+
+Lastly, a Top N chain can choose to exit ICS by issuing a `ConsumerRemovalProposal` and an Opt In chain can issue a transaction to stop the chain.
+After some period of time (e.g., provider's unbonding period), all state related to the stopped consumer chain can be removed. We
+keep track of the consumer chain's state for some period, so that we are able to punish validators for misbehaviour.
+
+Note that everything described so far and everything that follows applies to standalone chains as well.
+
+### From `chainID` to `consumerID`
+A hindrance in moving to Permissionless ICS is chain-id squatting. In a permissionless setting, anyone could issue a transaction
+to launch a consumer chain with a `chainID` that might already be used by some other consumer chain. This is a problem
+because in the current design the majority of stored state for a consumer chain is indexed using the `chainID` as the key (e.g.,
+see [key used to store client ids](https://github.com/cosmos/interchain-security/blob/v4.3.0/x/ccv/provider/types/keys.go#L233)).
+To tackle this problem, in Permissionless ICS, we introduce the `consumerID` that defines a consumer chain and is simply
```

Comment:
Could you briefly expand on the format of the consumer id? naively, depending on what "combination" means, there might be problems with the chain id "bleeding" into the sequence number, e.g.

neutron11 = chain_id ("neutron1") + sequence ("1")
but also
neutron11 = chain_id ("neutron") + sequence ("11)

Reply:
This is described later on where we write:
> The `consumerID`s corresponds to `len(chainID) | chainID | counter` and this way we can have multiple different `consumerID`s [...]

which is the common way of creating those keys. So, in the above example we would have two different keys: `8 | neutron1 | 1` and `7 | neutron | 11`.

Added the fact that the `consumerID` contains the `len(chainID`) in this part of the ADR as well.

Reply:
Perfect, thanks :)

Reply:
@p-offtermatt we should be preventing this globally by how we do our prefixes and stuff. I haven't been in the code for so long that I don't remember how we do it, but if we aren't any more i would be worried

Reply:
We could just use a counter but seems easier (e.g., for debugging purposes) to also have the `chainID` there.

Reply:
@mpoke Updated the ADR to just use a `counter`.